components/script/script_thread.rs

```diff
@@ -1919,6 +1926,9 @@ impl ScriptThread {
         ROUTER.route_ipc_receiver_to_mpsc_sender(ipc_timer_event_port,
                                                  self.timer_event_chan.clone());
 
+        let navigation_start_precise = precise_time_ns() as f64;
+        self.paint_time_metrics.set_navigation_start(navigation_start_precise);
```

Comment:
Reading this code made me file https://github.com/servo/servo/issues/17651. The fact that we wait to record this value until the first network response probably makes this PR more complicated than it would be otherwise.